aioredis/connection.py

```diff
@@ -373,7 +373,7 @@ def __init__(self, socket_read_size: int):
     def on_connect(self, connection: "Connection"):
         """Called when the stream connects"""
         self._stream = connection._reader
-        if self._buffer is None or self._stream is None:
+        if self._stream is None:
```

Comment:
I am not too sure what was the original intent behind this line, buffer only initialized after so it only can be `None` in this line.

Reply:
i believe we've had unexpected behavior here before, so keeping that there would be nice

Reply:
`self._stream is None` check is kept and test seems to be passing with it, The problem was `self._buffer is None` which always resulted in an exception. As it's not possible for it to be something other than `None` at this point.